### PR TITLE
adding options to download and upload artifacts

### PR DESCRIPTION
**Pull request by ric03uec**
- https://github.com/Shippable/kermit/issues/1448

### CODE REVIEW

File: execute/step/downloadArtifacts.js [hunk 1]
```diff
@@ -93,6 +93,7 @@ function _getStepArtifactUrl(bag, next) {
       }
 
       bag.stepArtifactUrl = artifactUrls.get;
+      bag.stepArtifactUrlOpts = artifactUrls.putOpts;
```
Comment (a-murphy) [inline execute/step/downloadArtifacts.js]: I know they were all the same in your last pull request, but why are you using the `get` URL with `putOpts`?

File: execute/step/downloadArtifacts.js [hunk 2]
```diff
@@ -123,7 +124,10 @@ function _getRunArtifactUrl(bag, next) {
       }
 
       bag.runArtifactUrl = artifactUrls.get;
+      bag.runArtifactUrlOpts = artifactUrls.putOpts;
```
Comment (a-murphy) [inline execute/step/downloadArtifacts.js]: Same question here.

File: execute/step/downloadArtifacts.js (hunk 1 above)
Comment (ric03uec) [reply execute/step/downloadArtifacts.js]: cop->paste error, fixed now

File: execute/step/downloadArtifacts.js (hunk 2 above)
Comment (ric03uec) [reply execute/step/downloadArtifacts.js]: fixed

**Pull request merged**